mempool/src/pool/tests/accumulator.rs

```diff
@@ -154,14 +169,148 @@ async fn collect_transactions(#[case] seed: Seed) -> anyhow::Result<()> {
     let returned_accumulator = mempool.collect_txs(Box::new(tx_accumulator)).unwrap();
     let collected_txs = returned_accumulator.transactions();
     assert_eq!(collected_txs.len(), 0);
 
-    let tx_accumulator = DefaultTxAccumulator::new(1, mempool.best_block_id());
+    let tx_accumulator = DefaultTxAccumulator::new(1, mempool.best_block_id(), DUMMY_TIMESTAMP);
     let returned_accumulator = mempool.collect_txs(Box::new(tx_accumulator)).unwrap();
     let collected_txs = returned_accumulator.transactions();
     assert_eq!(collected_txs.len(), 0);
     Ok(())
 }
+
+fn timelock_secs_after_genesis(n: u64) -> OutputTimeLock {
+    let mut rng = make_seedable_rng(Seed::from_u64(0));
+    let t0 = TestFramework::builder(&mut rng).build().genesis().timestamp();
+    OutputTimeLock::UntilTime(t0.add_int_seconds(n).unwrap())
+}
+
+#[rstest]
+#[trace]
+#[case::until_blk1(Seed::from_entropy(), OutputTimeLock::UntilHeight(1.into()), 0b1111)]
+#[trace]
+#[case::until_blk2(Seed::from_entropy(), OutputTimeLock::UntilHeight(2.into()), 0b1011)]
+#[trace]
+#[case::until_blk4(Seed::from_entropy(), OutputTimeLock::UntilHeight(4.into()), 0b1010)]
+#[trace]
+#[case::until_blk20(Seed::from_entropy(), OutputTimeLock::UntilHeight(20.into()), 0b0000)]
+#[trace]
+#[case::for_1blk(Seed::from_entropy(), OutputTimeLock::ForBlockCount(1), 0b0011)]
+#[trace]
+#[case::for_2blk(Seed::from_entropy(), OutputTimeLock::ForBlockCount(2), 0b0010)]
+#[trace]
+#[case::for_20blk(Seed::from_entropy(), OutputTimeLock::ForBlockCount(20), 0b0000)]
+#[trace]
+#[case::until_5s(Seed::from_entropy(), timelock_secs_after_genesis(5), 0b1111)]
+#[trace]
+#[case::until_10s(Seed::from_entropy(), timelock_secs_after_genesis(10), 0b1111)]
+#[trace]
+#[case::until_11s(Seed::from_entropy(), timelock_secs_after_genesis(11), 0b1011)]
+#[trace]
+#[case::until_30s(Seed::from_entropy(), timelock_secs_after_genesis(30), 0b1011)]
+#[trace]
+#[case::until_31s(Seed::from_entropy(), timelock_secs_after_genesis(31), 0b1010)]
+#[trace]
+#[case::until_500s(Seed::from_entropy(), timelock_secs_after_genesis(500), 0b0000)]
+#[trace]
+#[case::for_1s(Seed::from_entropy(), OutputTimeLock::ForSeconds(1), 0b0011)]
+#[trace]
+#[case::for_10s(Seed::from_entropy(), OutputTimeLock::ForSeconds(10), 0b0011)]
+#[trace]
+#[case::for_20s(Seed::from_entropy(), OutputTimeLock::ForSeconds(20), 0b0011)]
+#[trace]
+#[case::for_21s(Seed::from_entropy(), OutputTimeLock::ForSeconds(21), 0b0010)]
+#[trace]
+#[case::for_500s(Seed::from_entropy(), OutputTimeLock::ForSeconds(500), 0b0000)]
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn timelocked(#[case] seed: Seed, #[case] timelock: OutputTimeLock, #[case] expected: u32) {
+    // Unpack expected results:
+    let in_mempool_at0 = (expected & 0b1000) != 0;
+    let in_accumulator_at0 = (expected & 0b0100) != 0;
+    let in_mempool_at1 = (expected & 0b0010) != 0;
+    let in_accumulator_at1 = (expected & 0b0001) != 0;
```

Comment:
I can live with those being for tests... but I wish we used some proper struct for readability.

Reply:
Yea, I did not want the `#[case]` things to become too verbose. And quite a few of these combinations are legitimate outcomes, not easy to compress into a couple of enum variants.